### ISSUE TITLE
Data Views: Add subtle styles to interface components

### ISSUE THREAD

**Issue by jarekmorawski** [WordPress/gutenberg#66486]
Hi there! Did we ever consider adding subtle styles to our components? In particular, I'm thinking about dropdowns and input fields. The default styling works well when the interface's primary purpose is to edit data.

However, there are situations where a subtler style could be useful. The primary use case is inline editing in a Data View table or a large form where the user's primary goal is to _review_ the information and edit individual fields rather than whole sections.

A real-world use cases for subtler styles include the product variations list in WooCommerce or a field mapping table in a WordPress plugin I'm working on. Users who interact with these elements are primarily interested in reviewing the information. Displaying it in the editable state adds visual weight that may drive them away from the page.

![Image](https://github.com/user-attachments/assets/9665e5f9-d777-4e04-9bf3-41d784e9e423)

Components using the subtle style could act as interactive elements in a `readOnly` mode, with a highlight shown on hover. When clicked, the elements would transform into editable fields, like inputs and dropdowns.

https://github.com/user-attachments/assets/262ca454-7e00-408e-8de4-afb9d2af590d

There are existing implementations of such components in the [Atlassian Design System](https://atlassian.design/components/inline-edit/examples), [Greenhouse](https://greenhousedl.supernova-docs.io/latest/patterns/inputs/inline-edit/design-00cAoFFt), and other design systems that require a high degree of accessibility.

Here's an example from Atlassian's design system:

https://github.com/user-attachments/assets/5a4af73b-6791-46a4-a829-e3ce14163fdc

Is there a way to introduce something similar to the new wp-admin controls? If so, what needs to be done to ensure accessibility?

**Comment by joedolson** [comment]
From an accessibility perspective, I'm not particularly in favor of blurring the distinction between reading data and editing data. If a field is editable, it should be obvious that it's editable. I'd argue that if you really believe that a user doesn't want to edit a field in a particular context, then don't make it editable.

The problem with this for accessibility is kind of fundamental: how are you going to convey the information that this field is editable? For screen reader users it's pretty trivial: this is just styling, and they will probably be largely unaware that anything is different. 

However, for sighted users, this could be very disconcerting. I don't see any obvious way this could pass WCAG 1.4.11, which specifically requires that UI components have a 3:1 contrast with adjacent colors.

**Comment by anevins12** [comment]
The question of how you'd convey that this is editable is a good one.

If we look at how Atlassian achieve this, they're visually indenting the field. Perhaps this makes sense in isolation and in their interaction patterns.

For example, in Jira those fields may only appear in the sidebar when adding or editing an issue. This task is always the same and some people can learn that pattern a little bit easier for that reason. 

In our context, will WordPress users perceive that the indentation is the cue to an editable form field? 

Let's say we went ahead with the approach, we're still making the discovery of _type_ and _state_ information harder for sighted users. Instead of them being able to directly see that information in 1 step, they have multiple steps:
1. Identify the form control.
2. Navigate to and activate the form control.

For instance, if you had multiple fields and wanted to know which one was of type combobox, you'd have to hover and click into them.

I'm not advocating for the approach, although can appreciate the OP's want to reduce cognitive load. 

Additionally, I want to clarify the WCAG point @joedolson . A visual cue that does not rely on colour may still be sufficient to pass SC 1.4.11 Non-Text Contrast. This is of course subject to the interpretation of the auditor on whether they believe the indentation is a sufficient alternative to a customary border.

**Comment by joedolson** [comment]
@anevins12 I think that indentation could conceivably pass 1.4.11, but only if the *only instance* of that indentation is because the field is editable. I'm not sure that's a realistic requirement. To me, while it might pass that criteria, it's too subtle to be useful as an affordance.

**Comment by youknowriad** [comment]
For the form controls definition, I think we now have largely addressed this issue with the DataForm component (and its fields)
